gass/copy/source/globus_gass_copy_glob.c

```diff
@@ -2093,33 +2096,39 @@ globus_l_gass_copy_cksm_file(
     {
         count = GASS_COPY_CKSM_BUFSIZE;
     }
-    
-    fd = open(parsed_url.url_path, O_RDONLY);        
-    if(fd < 0)
-    {
-        result = globus_error_put(
-            globus_error_construct_string(
-                GLOBUS_GASS_COPY_MODULE,
-                GLOBUS_NULL,
-                "[%s]: error opening checksum file %s",
-                myname,
-                parsed_url.url_path));
-        goto error_fd;
+    result = globus_l_gass_copy_stat_file(url,&statbuf);
+    if( result != GLOBUS_SUCCESS){
+    	goto error_fd;
+    }
+    is_regular = statbuf.type == GLOBUS_GASS_COPY_GLOB_ENTRY_FILE;
+    if( is_regular ){
+    	fd = open(parsed_url.url_path, O_RDONLY);
+    	if(fd < 0)
+    	{
+    		result = globus_error_put(
+    				globus_error_construct_string(
+    						GLOBUS_GASS_COPY_MODULE,
+							GLOBUS_NULL,
+							"[%s]: error opening checksum file %s",
+							myname,
+							parsed_url.url_path));
+    		goto error_fd;
+    	}
+
+
+    	if (lseek(fd, offset, SEEK_SET) == -1)
+    	{
+    		result = globus_error_put(
+    				globus_error_construct_string(
+    						GLOBUS_GASS_COPY_MODULE,
+							GLOBUS_NULL,
+							"[%s]: error (%d) seeking checksum file %s",
+							myname,
+							errno,
+							parsed_url.url_path));
+    		goto error_seek;
+    	}
     }
-
-    if (lseek(fd, offset, SEEK_SET) == -1)
-    {
-        result = globus_error_put(
-            globus_error_construct_string(
-                GLOBUS_GASS_COPY_MODULE,
-                GLOBUS_NULL,
-                "[%s]: error (%d) seeking checksum file %s",
-                myname,
-                errno,
-                parsed_url.url_path));
-        goto error_seek;
-    }
-
     OpenSSL_add_all_algorithms();
```

Comment:
Maybe I missed something, but can't we move this whole block, up to line 2150 / 2159? In that case I think you can merge the two `is_regular` blocks into one and don't even need it any more, you can just use the `statbuf.type == GLOBUS_GASS_COPY_GLOB_ENTRY_FILE` as the expression in the single `if`.